src/MongoDB.Driver.Core/Core/Operations/ChangeStreamOperation.cs

```diff
@@ -382,59 +437,51 @@ private List<BsonDocument> CreateCombinedPipeline(BsonDocument changeStreamStage
             return combinedPipeline;
         }
 
-        private IAsyncCursor<RawBsonDocument> ExecuteAggregateOperation(IReadBinding binding, bool resuming, CancellationToken cancellationToken)
+        private IAsyncCursor<RawBsonDocument> ExecuteAggregateOperation(IReadBinding binding, CancellationToken cancellationToken)
         {
-            var aggregateOperation = CreateAggregateOperation(resuming);
+            var aggregateOperation = CreateAggregateOperation();
             return aggregateOperation.Execute(binding, cancellationToken);
         }
 
-        private Task<IAsyncCursor<RawBsonDocument>> ExecuteAggregateOperationAsync(IReadBinding binding, bool resuming, CancellationToken cancellationToken)
+        private Task<IAsyncCursor<RawBsonDocument>> ExecuteAggregateOperationAsync(IReadBinding binding, CancellationToken cancellationToken)
         {
-            var aggregateOperation = CreateAggregateOperation(resuming);
+            var aggregateOperation = CreateAggregateOperation();
             return aggregateOperation.ExecuteAsync(binding, cancellationToken);
         }
 
-        private ResumeStartValue GetEffectiveResumeStartValues(bool resuming)
+        private IBatchInfo GetBatchInfoOrThrow(IAsyncCursor<RawBsonDocument> cursor)
```

Comment:
No need for this helper method. Just cast directly to ICursorBatchInfo when needed.

Reply:
Done